File: com.rotovr.android.bleplugin/unitybleplugin/src/main/java/com/rotovr/unitybleplugin/model/RotoDataModel.java
package com.rotovr.unitybleplugin.model;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;

public class RotoDataModel implements Serializable {

    public RotoDataModel() {
        Mode = "";
        Angle = 0;
        TargetCockpit = 140;
        MaxPower = 100;
    }

    public RotoDataModel(byte[] data) {

        switch (data[2]) {
            case 0:
                Mode = "IdleMode";
                break;
            case 1:
                Mode = "Calibration";
                break;
            case 2:
                Mode = "HeadTrack";
                break;
            case 3:
                Mode = "FreeMode";
                break;
            case 4:
                Mode = "CockpitMode";
                break;
            case 5:
                Mode = "Error";
                break;
        }

        switch (data[5]) {
            case 0:
                Angle = data[6] & 0xFF;
                break;
            case 1:
                int angle = data[6] & 0xFF;
                Angle = (angle + 256);
                break;

        }
        TargetCockpit = data[9] & 0xFF;
        MaxPower = data[12] & 0xFF;
    }

    public String Mode;
    public int Angle;
    public int TargetCockpit;
    public int MaxPower;


    public boolean Compare(RotoDataModel model) {
        return Mode == model.Mode && Angle == model.Angle && TargetCockpit == model.TargetCockpit && MaxPower == model.MaxPower;
    }

    public String toJson() {
        JSONObject obj = new JSONObject();
        try {
            obj.put("Mode", Mode);
            obj.put("Angle", Angle);
            obj.put("TargetCockpit", TargetCockpit);
            obj.put("MaxPower", MaxPower);

            return obj.toString();
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return obj.toString();
    }
}

File: com.rotovr.sdk.package/Assets/com.rotovr.sdk/Runtime/Common/Model/RotoDataModel.cs
using System;

namespace com.rotovr.sdk
{
    /// <summary>
    /// Chair state model.
    /// </summary>
    [Serializable]
    public class RotoDataModel
    {
        
        /// <summary>
        /// Default state. FreeMod and 0 rotation.
        /// </summary>
        internal RotoDataModel()
        {
            Mode = ModeType.FreeMode.ToString();
            Angle = 0;
        }
        
        internal RotoDataModel(string mode, int angle, int targetCockpit, int maxPower)
        {
            Mode = mode;
            Angle = angle;
            TargetCockpit = targetCockpit;
            MaxPower = maxPower;
        }

        public string Mode { get; set; }

        public ModeType ModeType => EnumUtility.ParseOrDefault<ModeType>(Mode);
        
        public int Angle { get; set; }
        public int TargetCockpit { get; set; }
        public int MaxPower { get; set; }
    }
}